syntax = "proto3";

package code_execution;

option csharp_namespace = "Application.Common.Protos";

message File {
    string filename = 1;
    bytes content = 2;
}

message ExecutionJob {
    string job_id = 1;
    string room_id = 2;
    int32 language = 3;
    
    repeated File files = 4;
}

message ExecutionJobResult {
    string job_id = 1;
    string room_id = 2;

    bool error = 3;
    string error_message = 4;
    string output = 5;
    float execution_time = 6;
}